### PR TITLE
[Do Not Squash] GA Az.Monitor-preview

### PR DESCRIPTION
**Pull request by VeryEarly**
## Checklist

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Review by wyunchi-ms — DISMISSED** [review]

**Review by wyunchi-ms — APPROVED** [review]

**Review by wyunchi-ms — APPROVED** [review]

File: tools/Az/Az.psd1 [hunk 1]
```diff
@@ -98,7 +98,7 @@ RequiredModules = @(@{ModuleName = 'Az.Accounts'; ModuleVersion = '2.10.1'; },
                @{ModuleName = 'Az.MarketplaceOrdering'; RequiredVersion = '1.0.3'; }, 
                @{ModuleName = 'Az.Media'; RequiredVersion = '1.1.1'; }, 
                @{ModuleName = 'Az.Migrate'; RequiredVersion = '1.1.3'; }, 
-               @{ModuleName = 'Az.Monitor'; RequiredVersion = '3.1.0'; }, 
+               @{ModuleName = 'Az.Monitor'; RequiredVersion = '4.1.0'; },
```
Comment (dingmeng-xue) [inline tools/Az/Az.psd1:101]: Should not have change.

File: src/Monitor/Monitor/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -24,5 +24,5 @@
 [assembly: ComVisible(false)]
 [assembly: CLSCompliant(false)]
 [assembly: Guid("ed102280-3577-49bf-93dd-11b6e3a44a57")]
-[assembly: AssemblyVersion("3.1.0")]
-[assembly: AssemblyFileVersion("3.1.0")]
+[assembly: AssemblyVersion("4.1.0")]
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/Properties/AssemblyInfo.cs:27]: Should not have change. Please check all AssemblyInfo. Version should 3.1.0

File: src/Monitor/ActivityLogAlert.Autorest/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+[assembly: AssemblyTitle("Microsoft Azure Powershell - ActivityLogAlert")]
+[assembly: AssemblyCompany(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCompany)]
+[assembly: AssemblyProduct(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyProduct)]
+[assembly: AssemblyCopyright(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCopyright)]
+
+[assembly: ComVisible(false)]
+[assembly: CLSCompliant(false)]
+[assembly: Guid("68c9e5fe-3c10-46ed-b371-9a8c93ad49c7")]
+[assembly: AssemblyVersion("4.1.0")]
```
Comment (dingmeng-xue) [inline src/Monitor/ActivityLogAlert.Autorest/Properties/AssemblyInfo.cs:27]: Please update version

File: src/Monitor/DiagnosticSetting.Autorest/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+[assembly: AssemblyTitle("Microsoft Azure Powershell - DiagnosticSetting")]
+[assembly: AssemblyCompany(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCompany)]
+[assembly: AssemblyProduct(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyProduct)]
+[assembly: AssemblyCopyright(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCopyright)]
+
+[assembly: ComVisible(false)]
+[assembly: CLSCompliant(false)]
+[assembly: Guid("c0565a0c-7066-46c4-9090-0c88e7d16c5f")]
+[assembly: AssemblyVersion("4.1.0")]
```
Comment (dingmeng-xue) [inline src/Monitor/DiagnosticSetting.Autorest/Properties/AssemblyInfo.cs:27]: Please update version

File: src/Monitor/Monitor/Az.Monitor.psd1 [hunk 5]
```diff
@@ -119,7 +133,7 @@ CmdletsToExport = 'Get-AzMetricDefinition', 'Get-AzMetric', 'Remove-AzLogProfile
                'Remove-AzDataCollectionRuleAssociation'
 
 # Variables to export from this module
-# VariablesToExport = @()
+VariablesToExport = '*'
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/Az.Monitor.psd1:136]: should not change it.

File: src/Monitor/Monitor/Az.Monitor.psd1 [hunk 6]
```diff
@@ -151,14 +165,12 @@ PrivateData = @{
         # IconUri = ''
 
         # ReleaseNotes of this module
-        ReleaseNotes = '* Added breaking change warning messages for 
-    - ''ActivityLogAlert''
-    - ''DiagnosticSetting''
-    - ''ScheduledQueryRule''
-    - ''Autoscale'''
+        ReleaseNotes = '* Upgraded API version for DiagnosticSetting from 2017-05-01-preview to 2021-05-01-preview
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/Az.Monitor.psd1:168]: It should not change.

File: src/Monitor/Monitor/Az.Monitor.psd1 [hunk 7]
```diff
@@ -151,14 +165,12 @@ PrivateData = @{
...
+* Upgraded API version for ScheduledQueryRule from 2018-04-16 to 2021-08-01
+* Upgrade API version for Autoscale from 2015-04-01 to 2022-10-01'
 
         # Prerelease string of this module
-        # Prerelease = ''
+        #Prerelease = 'preview'
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/Az.Monitor.psd1:173]: should not change it.

File: src/Monitor/Monitor/ChangeLog.md [hunk 8]
```diff
@@ -19,6 +19,42 @@
...
+    - `New-AzActionGroup` replaced by `New-AzActivityLogAlertActionGroupObject`
+* [BreakingChange] Upgraded API version for DiagnosticSetting from 2017-05-01-preview to 2021-05-01-preview
+    - `Get-AzDiagnosticSettingCategory`
+    - `Get-AzDiagnosticSetting`
+    - `New-AzDiagnosticSetting`
+    - `Remove-AzDiagnosticSetting`
+    - `Set-AzDiagnosticSetting`
+    - `New-AzDiagnosticDetailSetting` replaced by `New-AzDiagnosticSettingLogSettingsObject` and `New-AzDiagnosticSettingMetricSettingsObject`
+    - `Get-AzSubscriptionDiagnosticSettingCategory` replaced by `Get-AzEventCategory`
+* [BreakingChange] Upgrade API version for Autoscale from 2015-04-01 to 2022-10-01
```
Comment (dingmeng-xue) [inline src/Monitor/Monitor/ChangeLog.md:38]:
```suggestion
* [BreakingChange] Upgraded API version for Autoscale from 2015-04-01 to 2022-10-01
```

File: src/Monitor/ScheduledQueryRule.Autorest/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+[assembly: AssemblyTitle("Microsoft Azure Powershell - ScheduledQueryRule")]
+[assembly: AssemblyCompany(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCompany)]
+[assembly: AssemblyProduct(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyProduct)]
+[assembly: AssemblyCopyright(Microsoft.WindowsAzure.Commands.Common.AzurePowerShell.AssemblyCopyright)]
+
+[assembly: ComVisible(false)]
+[assembly: CLSCompliant(false)]
+[assembly: Guid("60e17e57-5f4f-4fa2-809f-6d1b00f496a9")]
+[assembly: AssemblyVersion("4.1.0")]
```
Comment (dingmeng-xue) [inline src/Monitor/ScheduledQueryRule.Autorest/Properties/AssemblyInfo.cs:27]: should correct version.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**